build.gradle.kts

```diff
@@ -1,9 +1,49 @@
+import net.pwall.json.kotlin.codegen.gradle.JSONSchemaCodegenPlugin
+import net.pwall.json.kotlin.codegen.gradle.JSONSchemaCodegenTask
+
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+    dependencies {
+        // A.t.m. we need our patched version of json-kotlin-schema-codegen here, locally supplied via "mvn install".
+        classpath("net.pwall.json:json-kotlin-schema-codegen:0.108")
+        classpath("net.pwall.json:json-kotlin-gradle:0.107")
+    }
+}
+
 plugins {
     id("buildlogic.kotlin-common-conventions")
 }
 
 dependencies {
     // This merges all our individual kover results into the root project
     kover(project(":csaf-import"))
-    kover(project(":csaf-schema-codegen"))
-}
\ No newline at end of file
+}
+
+// Create and register ExecutionService which enforces serial execution of assigned tasks.
+abstract class SerialExecutionService : BuildService<BuildServiceParameters.None>
+val serialExecutionService =
+    gradle.sharedServices.registerIfAbsent("serialExecution", SerialExecutionService::class.java) {
+        this.maxParallelUsages.set(1)
+    }
+
+apply<JSONSchemaCodegenPlugin>()
+
+// Configure gradle caching manually for json-kotlin-gradle, as the plugin seems to lack support for it.
+tasks.withType(JSONSchemaCodegenTask::class) {
+    inputs.file("src/main/resources/codegen-config.json").withPathSensitivity(PathSensitivity.RELATIVE)
+    inputs.dir("src/main/resources/schema").withPathSensitivity(PathSensitivity.RELATIVE)
+    outputs.dir("build/generated-sources/kotlin")
+}
+
+allprojects {
```

Comment:
there should not be the need for an `allprojects` here, information like that should be moved into the build-logic.